File: 242.valid-anagram.c
#define CHARSET_SIZE 26

bool isAnagram(char * s, char * t){
    int i;
    int chcounts[CHARSET_SIZE] = { 0 };

    i = 0;
    while (s[i] != '\0') {
        chcounts[s[i]-'a'] += 1;
        i++;
    }

    i = 0;
    while (t[i] != '\0') {
        chcounts[t[i]-'a'] -= 1;
        i++;
    }

    for (i = 0; i < CHARSET_SIZE; ++i) {
        if (chcounts[i])
            return 0;
    }
    return 1;
}

File: 242.valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        chcounts = {}
        for ch in s:
            if ch not in chcounts:
                chcounts[ch] = 0
            chcounts[ch] += 1

        for ch in t:
            if ch not in chcounts:
                return False
            chcounts[ch] -= 1

        for count in chcounts.values():
            if count:
                return False

        return True
